File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineExam6And7April2019/02.GodzillaVsKong.js
function godzillaVsKong(input) {
    let filmBudget = Number(input[0]);
    let numberForExtras = Number(input[1]);
    let clothingPriceForOne = Number(input[2]);

    let decorPrice = filmBudget * 0.10;
    let clothingPriceForAll = numberForExtras * clothingPriceForOne;

    if(numberForExtras > 150) {
        clothingPriceForAll = clothingPriceForAll * 0.90;
    }

    let totalNeededMoney = clothingPriceForAll + decorPrice;

    let moneyDiff = Math.abs(filmBudget - totalNeededMoney);

    if(filmBudget < totalNeededMoney) {
        console.log('Not enough money!');
        console.log(`Wingard needs ${moneyDiff.toFixed(2)} leva more.`);
    } else {
        console.log('Action!');
        console.log(`Wingard starts filming with ${moneyDiff.toFixed(2)} leva left.`);
    }
}

godzillaVsKong(["20000", 
"120",
"55.5"])

godzillaVsKong(["15437.62",
"186",
"57.99"])

godzillaVsKong(["9587.88",
"222",
"55.68"])

File: Programming_Basics_with_Python/programming_basics_online_exam_6_and_7_april_2019/02._godzilla_vs_kong.py
def film_fee(film_budget, people_count, clothes_price_per_person, film_set_price, have_discount_for_more_people, discount_amount):
    people_clothes_price = people_count * clothes_price_per_person
    if people_count > have_discount_for_more_people:
        people_clothes_price = people_clothes_price * discount_amount

    total_fee = people_clothes_price + film_set_price
    money_difference = film_budget - total_fee
    if money_difference >= 0:
        return f'Action!\nWingard starts filming with {money_difference:.2f} leva left.'
    else:
        return f'Not enough money!\nWingard needs {abs(money_difference):.2f} leva more.'


film_budget = float(input())
people_count = int(input())
clothes_price_per_person = float(input())
film_set_price = film_budget * 0.1
have_discount_for_more_people = 150
discount_amount = 0.9


print(film_fee(film_budget, people_count, clothes_price_per_person, film_set_price, have_discount_for_more_people, discount_amount))
